### ISSUE TITLE
Empty Vertex Array, procedurally generate vertex data

### ISSUE THREAD

**Issue by habemus-papadum** [tsherif/picogl.js#105]
Hi -- I have a question about the use case where I would like to use an empty vertex array but generate vertex data in shader using `gl_vertexID`.  For example the following gist currently creates a triangle:

vert-shader:
````
#version 300 es
        out vec3 vColor;
        void main() {
            vColor = vec3(float(gl_VertexID)/3.0, float((gl_VertexID + 2) % 3)/3.0, 0.4);
            gl_Position = vec4(2.f * float(uint(gl_VertexID) % 2u) - 1.f, 2.f * float(uint(gl_VertexID) / 2u) - 1.f, 0.0, 1.0);
        }
````
javascript fragment:
````
    var empty = app.createVertexArray()
    empty.numElements = 3
    var drawCall = app.createDrawCall(program, empty);
````

The `empty.numElements = 3` works but feels risky.  Also, it seems the a better place for this would be to modify `createDrawCall` to  `createDrawCall(p,a,primitive,numEl,numInst)` where the `num...` vars  override the values inferred from buffer size (with some sort of sanity checking - the details of which are not clear to me.)

I'm curious if there are thoughts and/or interest about this usage -- let me know and I'll try to synthesize a PR. nehal

**Comment by tsherif** [comment]
I like this idea. How about if we modify the VertexArray class so that it supports setting numElements and numInstances directly via the constructor, and doesn't actually create the GL VAO until the first time something is bound to it (either in the `attributeBuffer` or `indexBuffer` methods).

So the API would really have to change much. E.g. @habemus-papadum's example would look like this:
```
var empty = app.createVertexArray(3);
var drawCall = app.createDrawCall(program, empty);
```

And an instanced version could just have an optional second argument:
```
var empty = app.createVertexArray(3, 4);
var drawCall = app.createDrawCall(program, empty);
```

**Comment by habemus-papadum** [comment]
Sounds good to me.

**Comment by tsherif** [comment]
FIxed by #107
